contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5BatchReader.java

```diff
@@ -410,8 +390,22 @@ public boolean next() {
     return true;
   }
 
+  @Override
+  public void close() {
+    AutoCloseables.closeSilently(hdfFile, reader);
+    /*
+     * The current implementation of the HDF5 reader creates a temp file which needs to be removed
+     * when the batch reader is closed. A possible future functionality might be to used to.
+     */
+    boolean result = hdfFile.getFile().delete();
```

Comment:
There was a comment above about reading from a byte array. We can't read from the original file? Are we reading from a byte array or a temp file?

If a temp file, where is the temp file located? In the system temp dir, or in Drill's configured temp dir?

Reply:
https://github.com/jamesmudd/jhdf/blob/3136bba6d5c6d6956bea006865c7241b6ddc6f25/jhdf/src/main/java/io/jhdf/HdfFile.java#L167-L169
In the above code, these temporary files are deleted only before application exits.